### ISSUE TITLE
Paths to library directories in Overrides.toml

### ISSUE THREAD

**Issue by vchuravy** [JuliaLang/Pkg.jl#1704]
@staticfloat I have been working with `Overrides.toml` the last few days and I have encountered several minor nuisances, related to
libraries.

1. Overriding dependencies of stdlibs. E.g. I want OpenBLAS_jll to pickup the Julia OpenBLAS that I built instead of trying to load the artifact. Those libraries are in `$prefix/lib/julia`
2. Overriding `CompilerSupportLibraries_jll` I wanted to point it at `${gcc-prefix}/usr/lib64`

This of course is due to my wonderful live on RHEL 7.6

```
julia> using CompilerSupportLibraries_jll
ERROR: InitError: could not load library "/home/vchuravy/.julia/artifacts/59ca0cd093f9b05d3c6365c1147e7d908fa6954a/lib/libstdc++.so"
/lib64/ld64.so.2: version `GLIBC_2.22' not found (required by /home/vchuravy/.julia/artifacts/59ca0cd093f9b05d3c6365c1147e7d908fa6954a/lib/libstdc++.so)
```

**Comment by StefanKarpinski** [comment]
cc @staticfloat

**Comment by KristofferC** [comment]
Could you give a bit more details about what you tried and what the actual problems were? So far, it isn't (at least to me) clear what the problem actually is.

**Comment by vchuravy** [comment]
As an example the `Overrides.toml` I have is:

```
[bea87d4a-7f5b-5778-9afe-8cc45184846c]
SuiteSparse = "/home/software/julia/1.3.0/local/share/julia"

[efe28fd5-8261-553b-a9e1-b2916fc3738e]
OpenSpecFun = "/home/software/julia/1.3.0/local/share/julia"

[4536629a-c528-5b80-bd46-f80d51c5b363]
OpenBLAS = "/home/software/julia/1.3.0/local/share/julia"
```

Julia vendors the libraries it ships with so they go into `/home/software/julia/1.3.0/usr/lib/julia`. The reason why I am pointing to `/home/software/julia/1.3.0/local/share/julia` is because I created a softlink from `/home/software/julia/1.3.0/local/share/julia/lib` to `/home/software/julia/1.3.0/usr/lib/julia`.

What I would have wanted to say is:
```
[4536629a-c528-5b80-bd46-f80d51c5b363]
OpenBLAS.libdir = "/home/software/julia/1.3.0/usr/lib/julia"
```

Similarly with CompilerSupportLibraries_jll I wanted to point it at the lib suffix there is `lib64` and not `lib` so I need to create yet another softlink instead of just specifing the path to look in directly.

**Comment by KristofferC** [comment]
My guess is https://github.com/JuliaLang/julia/issues/33973 might help with some of that,.

**Comment by vchuravy** [comment]
Yes and no. My problem indirectly is that BB requires a glibc that is newer than what ever RedHat 7.6 is shipping with. I am working on remedying this separately. But it won't change the fact that library directories for Overrides.toml will have different suffixes than just `lib`, `lib64` is rather common.

**Comment by barche** [comment]
Same problem trying to override Qt_jll on Fedora, only workaround I could find was to make a symlink.

**Comment by KristofferC** [comment]
Artifacts is not really a component of Pkg anymore and I think the Overrides.toml is deprecated for Preference usage? Feel free to reopen at either the julia repo or Preferences.jl.
